alerting-config-service-impl/src/main/java/org/hypertrace/alerting/config/service/EventConditionStore.java

```diff
@@ -1,117 +1,82 @@
 package org.hypertrace.alerting.config.service;
 
-import static org.hypertrace.alerting.config.service.AlertingConfigServiceConstants.ALERTING_CONFIG_NAMESPACE;
-import static org.hypertrace.alerting.config.service.AlertingConfigServiceConstants.ALERTING_EVENT_CONDITION_CONFIG_RESOURCE_NAME;
-
 import com.google.protobuf.InvalidProtocolBufferException;
 import com.google.protobuf.Value;
 import io.grpc.Channel;
-import java.util.List;
+import java.util.Optional;
 import java.util.UUID;
-import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
 import org.hypertrace.alerting.config.service.v1.EventCondition;
+import org.hypertrace.alerting.config.service.v1.EventConditionData;
+import org.hypertrace.alerting.config.service.v1.EventConditionData.ConditionCase;
 import org.hypertrace.alerting.config.service.v1.NewEventCondition;
-import org.hypertrace.alerting.config.service.v1.NewEventCondition.ConditionCase;
+import org.hypertrace.config.objectstore.IdentifiedObjectStore;
 import org.hypertrace.config.proto.converter.ConfigProtoConverter;
 import org.hypertrace.config.service.v1.ConfigServiceGrpc;
-import org.hypertrace.config.service.v1.ConfigServiceGrpc.ConfigServiceBlockingStub;
-import org.hypertrace.config.service.v1.DeleteConfigRequest;
-import org.hypertrace.config.service.v1.GetAllConfigsRequest;
-import org.hypertrace.config.service.v1.UpsertConfigRequest;
-import org.hypertrace.core.grpcutils.client.GrpcClientRequestContextUtil;
 import org.hypertrace.core.grpcutils.client.RequestContextClientCallCredsProviderFactory;
 import org.hypertrace.core.grpcutils.context.RequestContext;
 
-public class EventConditionStore {
+@Slf4j
+public class EventConditionStore extends IdentifiedObjectStore<EventCondition> {
 
-  private final ConfigServiceBlockingStub configServiceBlockingStub;
+  private static final String ALERTING_EVENT_CONDITION_CONFIG_RESOURCE_NAME =
+      "alertingEventConditionConfig";
+  private static final String ALERTING_CONFIG_NAMESPACE = "alerting-v1";
 
   public EventConditionStore(Channel configChannel) {
-    this.configServiceBlockingStub =
+    super(
         ConfigServiceGrpc.newBlockingStub(configChannel)
             .withCallCredentials(
-                RequestContextClientCallCredsProviderFactory.getClientCallCredsProvider().get());
+                RequestContextClientCallCredsProviderFactory.getClientCallCredsProvider().get()),
+        ALERTING_CONFIG_NAMESPACE,
+        ALERTING_EVENT_CONDITION_CONFIG_RESOURCE_NAME);
   }
 
   public EventCondition createEventCondition(
       RequestContext requestContext, NewEventCondition newEventCondition) {
     EventCondition.Builder builder = EventCondition.newBuilder();
-    if (newEventCondition.getConditionCase() == ConditionCase.METRIC_ANOMALY_EVENT_CONDTION) {
-      builder.setMetricAnomalyEventCondtion(newEventCondition.getMetricAnomalyEventCondtion());
+    if (newEventCondition.getEventConditionData().getConditionCase()
```

Comment:
As far as separation of concern goes, you'd ideally want validation and object building to happen outside the store/dao layer.

Reply:
Building the object in the impl class now.